src/components/utilities/Drop.tsx

```diff
@@ -60,6 +60,14 @@ interface DropProps {
 	children: React.ReactNode | React.ReactNode[];
 }
 
+function isOverlayElement(element: React.DragEvent<Element> | DragEvent | null): boolean {
+	const target = element?.target;
+	if (target instanceof HTMLElement) {
+		return !!target.outerHTML?.toLowerCase().includes(OVERLAY_ELEMENT_IDENTIFIER);
```

Comment:
I think that, in this context, the usage of the _includes_ method is not 100% safe nor efficient.
The outerHTML contains the tag itself and all of its content. If the target isn't a _overlay-element_ but it contains it (or it contains the "overlay-element" string) the _isOverlayElement_ return true anyway.
Moreover if the target contains a large content (let's say a _img_ with a base64 content source) the _includes_ comparison will take some time.
Probably we can achieve better results by checking the _target.tagName_ property

Reply:
I did struggle to find a solution to detect the element, and I agree with you that an html property read on the outerHTML attribute is not fail proof. but in absence of alternatives I think it's reasonable enough for our goal, tagName is not viable as it will always return DIV, so it won't help us.

in terms of safety, the corner cases that you are describing are in my view very unlikely to occur, I can however change the OVERLAY_ELEMENT_IDENTIFIER value to a more complex string?

i get your point in the search speed, I can maybe replace the includes with a regex test, which could have a somehow better performance?

Reply:
Whoops, I missed the fact that searched string is the value of an attribute and not the element tag name, sorry about that.
So, instead of the tag name, the specific attribute should be inspected. This can be achieved by using the _attributes_ property of the element.
Something like:
`target.attributes.getNamedItem('type') `
should return the value of the attribute that we need to check

Reply:
that is returning null, the value is not being read. the attributes property does not have a type attribute